plugins/device/deviceutils/fstab.go

```diff
@@ -0,0 +1,173 @@
+package deviceutils
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"syscall"
+)
+
+// FstabMount represents entry in Fstab file
+type FstabMount struct {
+	Device           string
+	MountPoint       string
+	FilesystemFormat string
+	MountOptions     string
+	DumpValue        string
+	FsckOption       string
+}
+
+// Fstab represents Fstab entries
+type Fstab struct {
+	Filename string
+	Mounts   []FstabMount
+}
+
+func lock(filename string) (*os.File, error) {
+	lockfile, err := os.Create(filename + ".lock")
+	if err != nil {
+		return nil, err
+	}
+	defer lockfile.Close()
```

Comment:
The file is closed when the function returns. Doesn't closing file description implicitly unlock ?

Reply:
yes, will remove this